packages/trip-form/src/__mocks__/modes-en.ts

```diff
@@ -194,4 +194,27 @@ const commonModes = {
   ]
 };
 
+// This variant of the data adds an option with a company as the first
+// option in the "CAR" category that has mixed modes ("CAR_PARK", "CAR_HAIL")
+export const modesWithCompanyFirstMixedCategory = {
+  ...commonModes,
+  categories: commonModes.categories.map(c => {
+    const nextCategory = { ...c };
+
+    if (c.id === "CAR") {
+      nextCategory.options = [
+        {
+          company: "LYFT",
+          mode: "CAR_HAIL",
+          label: "Lyft",
+          url: "https://www.lyft.com"
```

Comment:
Is `url` important for this PR? If so, add an optional `url` to type `Category` in `src/TripOptions/types.d.ts`, otherwise delete this `url` attribute.

Reply:
> Please consider comments based on code before you added the exceptions to type checking. I think enforcing types even on stories helps keeping the mock data and the various data types in sync.

Hi @binh-dam-ibigroup, thank you for reviewing this, I hadn't seen your comments at the time I adjusted the typescript config and I'm glad to revert that change if you think that's the way to go, but let me explain a little further what I've found:

With respect to this comment, you're looking at the wrong type definition, it's `ModeOption` that being validated against, not `Category` and the former already has an [optional `url` property](https://github.com/opentripplanner/otp-ui/blob/master/packages/trip-form/src/TripOptions/types.d.ts#L12).  The problem is that the typescript checks seem to be applying the types from `trip-form/src/types.ts` (which has similar, but not identical types defined) instead of `trip-form/src/TripOptions/types.d.ts` and this mock data is designed to work with the latter.  I'm a bit confused about how typescript is matching the mock data to types, its seems to kind of be making sort of an educated guess, but that doesn't work out in this instance.

Also, when the mock data is declared explicitly like `commonModes` [here](https://github.com/opentripplanner/otp-ui/blob/master/packages/trip-form/src/__mocks__/modes-en.ts#L6)  typescript doesn't seem to be doing any checks on it, if it were [this](https://github.com/opentripplanner/otp-ui/blob/master/packages/trip-form/src/__mocks__/modes-en.ts#L64) options section should raise the same violations as the new mock object, as it's in the same format as the Lyft option that I add in the variant.  It seems that type checking is only being done on the mocks when properties are *re*assigned